javascript/ql/lib/semmle/javascript/frameworks/NoSQL.qll

```diff
@@ -638,6 +659,10 @@ private module MarsDB {
       )
     }
 
+    override DataFlow::Node getAResult() {
+      PromiseFlow::loadStep(this.(DataFlow::SourceNode).getALocalUse(), result, Promises::valueProp())
```

Comment:
If you add `DataFlow::CallNode` to the `MarsDBAccess` class, then this inline cast becomes redundant. 

There is a bunch of others that are similar, I haven't added a comment on the rest.

Reply:
thx!